### ISSUE TITLE
Intermittent error in useChat when state is 'partial-call'

### ISSUE THREAD

**Issue by zachtball** [vercel/ai#3196]
### Description

I am using the useChat hook to call my chat api route handler in NextJS. I have datadog logging any errors in useChat's `onError` property. I randomly get errors logged that look like this:

```
Error: ToolInvocation must have a result: {"state":"partial-call","toolCallId":"call_id","toolName":"myToolName","args":{"message":"my arg.","question":"my arg"}}
```

It is always when `state` is `partial-call`.

### Code example

```
  const { messages, input, handleInputChange, append, isLoading, setMessages } = useChat({
    maxSteps: 1,
    onError: (error) => {
      logger.error(`chat error; messages: ${JSON.stringify(messages)}`, error);
    }
  });
```

api route:
```
...

const result = await streamText({
      model: openai('gpt-4o'),
      toolChoice,
      maxTokens: 2400,
      temperature,
      frequencyPenalty,
      presencePenalty,
      experimental_toolCallStreaming,
      messages: convertToCoreMessages(convertedMessages),
      tools: {
        myToolName: tool({
          description: 'Generate some params.',
          parameters: myParams,
          execute: async (params) => {
            return params;
          },
        }),
       // ... some other tools
      },
    });

    return result.toDataStreamResponse();
```

**Comment by cmaycumber** [comment]
Running into a similar issue. Only have the issue when experimental partial-calls are enabled.

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/pull/3320

**Comment by lgrammel** [comment]
Reverting since it broke useChat.

**Comment by lgrammel** [comment]
I think this is the result of tool calls not having proper tool results. In `ai@3.4.25` we shipped a fix in that area. Please re-open if you still encounter this issue with `ai@3.4.25+`

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete failure in existing streamed tool-call behavior, and the maintainer identified and shipped a fix for the underlying result handling.

### Evidence
- useChat intermittently invokes onError with “ToolInvocation must have a result” when a streamed tool invocation is in the valid intermediate `partial-call` state.
- Another user confirmed the failure occurs when experimental partial tool calls are enabled.
- A maintainer linked a proposed fix and later stated that `ai@3.4.25` shipped a fix for tool calls lacking proper results.
